src/react-native-action-sheet.js

```diff
@@ -0,0 +1,29 @@
+/* @flow strict-local */
+import type { ComponentType, ElementConfig } from 'react';
+import { connectActionSheet as connectActionSheetInner } from '@expo/react-native-action-sheet';
+
+import type { BoundedDiff } from './generics';
+
+/* eslint-disable flowtype/generic-spacing */
+
+export type ShowActionSheetWithOptions = (
+  { options: string[], cancelButtonIndex: number },
+  (number) => void,
+) => void;
+
+/**
+ * Exactly like the `connectActionSheet` in
+ *   `react-native-action-sheet` upstream, but more typed.
```

Comment:
action sheet: Add type wrapper for `connectActionSheet`.
    
    `connectActionSheet`'s poor typing is one thing stopping
    `MessageList`'s props from being type checked.
    
    Tightening up the types of those props will take some work, starting
    from sorting out the various hacks the search-messages screen uses
    when it passes props with names that match the names we assign to
    data from react-redux.
    
    Like we did for react-native-safe-area-context in 8f569641f, use a
    type-wrapper instead of a libdef so that we can use our handy
    `BoundedDiff` type.

Excellent :smile: 

I think the "Tightening up the types … will take some work" graf can pretty much be dropped -- it mostly describes the work that's now done in the earlier commits of this branch. If a version of it is kept, it should be updated accordingly.

Reply:
Ah, right—looks like this commit started its life much earlier in the branch, then I moved it later without updating the commit message.